### PR TITLE
Mark DBO test as flaky on b200 for Distributed B200 test

### PR DESCRIPTION
**Pull request by dougbtv**
## Purpose

Mark DBO test that is failing the expected accuracy as flaky until an SME can diagnose the issue at hand.

See also: https://buildkite.com/vllm/ci/builds/41525#019ade7c-198d-42ee-b5ba-4f7636f08e32/72-1786 -- which is introduced after #29822 fixes the nccl header issues.

## Test Plan

Plain CI run, especially passing on the Distributed B200 tests

## Test Result

Confirmed `Distributed Tests (B200)` "passes" -- test shows as green and pytest is properly marked xfail with appropriate outpu. See associated test: https://buildkite.com/vllm/ci/builds/41643#019ae11c-e39e-4aa7-8fd4-6c9830e38a34

e.g.

```
XFAIL (Temporary: DBO accuracy unstable on Blackwell (doesn't meet ex...)
```

---
- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results

### CODE REVIEW

File: tests/v1/distributed/test_dbo.py [hunk 1]
```diff
@@ -9,10 +9,15 @@
...
 
 import pytest
+import torch
 
 from tests.evals.gsm8k.gsm8k_eval import evaluate_gsm8k
 from tests.utils import RemoteOpenAIServer
 
+# Detect Blackwell / B200 (compute capability 10.x)
+cap = torch.cuda.get_device_capability(0)
+IS_BLACKWELL = cap[0] >= 10
```
[the remark this answers is not in the document]
Comment (dougbtv) [reply tests/v1/distributed/test_dbo.py]: got it, thx

**Comment by dougbtv** [comment]
manual CI run: https://buildkite.com/vllm/ci/builds/41643

**Review by robertgshaw2-redhat — APPROVED** [review]
We should defintiely run this down though? Is someone assigned to look into it?
